tests/test_flow/test_persistence_fuzz.py

```diff
@@ -270,21 +323,40 @@ def run(self, n_iterations):
                     self.model.expected_entity_value(affected_name)
                 )
 
-            if make_func_change or self._versioning_mode == 'auto':
-                expected_called_names = self.model.entity_names(
-                    downstream_of=updated_name,
-                    upstream_of=affected_names,
-                )
-                assert (
-                    list(sorted(self.model.called_entity_names())) ==
-                    list(sorted(expected_called_names))
-                )
-            else:
-                assert len(self.model.called_entity_names()) == 0
+            actual_called_names = self.model.called_entity_names()
+            expected_called_names = self._expected_called_names(
+                updated_name, affected_names, make_func_change)
+            assert set(actual_called_names) == expected_called_names
 
     def _random_bool(self):
         return self._random.choice([True, False])
 
+    def _random_bool_with_weighted_probability(self, prob_true):
+        assert 0.0 <= prob_true <= 1.0
+        return choice([True, False], p=[prob_true, 1 - prob_true])
+
+    def _expected_called_names(
```

Comment:
As we discussed on Slack, this function signature feels awkward to me, but I don't have any better ideas either.